From c41144ce0d72cb19d347a46ef9386f257f14fb58 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 28 Oct 2012 23:19:39 +0100
Subject: [PATCH] Add multilib support

pango.modules list modules specific to a host architecture.
Add host triplet in path so that machine able to run multiple
triplet have a proper file per pango library.
---
 pango/Makefile.am   | 1 +
 pango/pango-utils.c | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/pango/Makefile.am b/pango/Makefile.am
index ab092c9..558108b 100644
--- a/pango/Makefile.am
+++ b/pango/Makefile.am
@@ -17,6 +17,7 @@ INCLUDES =						\
 	-DPANGO_ENABLE_BACKEND				\
 	-DPANGO_ENABLE_ENGINE				\
 	-DSYSCONFDIR=\"$(sysconfdir)\"			\
+	-DHOST=\"$(host_triplet)\"			\
 	-DLIBDIR=\"$(libdir)\"				\
 	-I$(top_srcdir)					\
 	-I$(top_builddir)				\
diff --git a/pango/pango-utils.c b/pango/pango-utils.c
index 18ffa26..3b88f15 100644
--- a/pango/pango-utils.c
+++ b/pango/pango-utils.c
@@ -738,9 +738,9 @@ pango_get_sysconf_subdirectory (void)
 #else
       const char *sysconfdir = g_getenv ("PANGO_SYSCONFDIR");
       if (sysconfdir != NULL)
-	tmp_result = g_build_filename (sysconfdir, "pango", NULL);
+	tmp_result = g_build_filename (sysconfdir, "pango", HOST, NULL);
       else
-	tmp_result = SYSCONFDIR "/pango";
+	tmp_result = SYSCONFDIR "/pango/" HOST;
 #endif
       g_once_init_leave((gsize*)&result, (gsize)tmp_result);
     }
-- 
1.7.12.4

